src/lpython/parser/semantics.h

```diff
@@ -283,76 +284,131 @@ static inline withitem_t *WITH_ITEM(Allocator &al, Location &l,
 #define WITH(items, body, l) make_With_t(p.m_a, l, \
         items.p, items.size(), STMTS(body), body.size(), nullptr)
 
-static inline arg_t *FUNC_ARG(Allocator &al, Location &l, char *arg, expr_t* e) {
-    arg_t *r = al.allocate<arg_t>();
-    r->loc = l;
-    r->m_arg = arg;
-    r->m_annotation = e;
-    r->m_type_comment = nullptr;
+static inline Arg *FUNC_ARG(Allocator &al, Location &l, char *arg,
+        expr_t* e, expr_t* defaults) {
+    Arg *r = al.allocate<Arg>();
+    r->_arg.loc = l;
+    r->_arg.m_arg = arg;
+    r->_arg.m_annotation = e;
+    r->_arg.m_type_comment = nullptr;
+    if(defaults){
+        r->def = true;
+    }
+    r->defaults = defaults;
     return r;
 }
 
 static inline Args *FUNC_ARGS(Allocator &al, Location &l,
-        arg_t* m_posonlyargs, size_t n_posonlyargs,
-        arg_t* m_args, size_t n_args,
-        arg_t* m_vararg, size_t n_vararg,
-        arg_t* m_kwonlyargs, size_t n_kwonlyargs,
-        expr_t** m_kw_defaults, size_t n_kw_defaults,
-        arg_t* m_kwarg, size_t n_kwarg,
-        expr_t** m_defaults, size_t n_defaults) {
+        Arg* /*m_posonlyargs*/, size_t /*n_posonlyargs*/,
+        Arg** m_args, size_t n_args,
+        Arg* m_vararg, size_t n_vararg,
+        Arg** m_kwonlyargs, size_t n_kwonlyargs,
+        expr_t** /*m_kw_defaults*/, size_t /*n_kw_defaults*/,
+        Arg* m_kwarg, size_t n_kwarg) {
     Args *r = al.allocate<Args>();
+    Vec<arg_t> _m_args;
+    _m_args.from_pointer_n(nullptr, 0);
+    Vec<arg_t> _m_vararg;
+    _m_vararg.from_pointer_n(nullptr, 0);
+    Vec<arg_t> _m_kwonlyargs;
+    _m_kwonlyargs.from_pointer_n(nullptr, 0);
+    Vec<arg_t> _m_kwarg;
+    _m_kwarg.from_pointer_n(nullptr, 0);
+    Vec<expr_t*> defaults;
+    defaults.from_pointer_n(nullptr, 0);
+    if(n_args > 0) {
+        _m_args.reserve(al, n_args);
+        defaults.reserve(al, n_args);
+        for(size_t i = 0; i < n_args; i++) {
+            _m_args.push_back(al, m_args[i]->_arg);
+            if(m_args[i]->def) {
+                defaults.push_back(al, m_args[i]->defaults);
+            }
+        }
+    }
+    if(n_vararg == 1) {
+        _m_vararg.reserve(al, n_vararg);
+        defaults.reserve(al, n_vararg);
+        _m_vararg.push_back(al, m_vararg->_arg);
+        if(m_vararg->def) {
+            defaults.push_back(al, m_vararg->defaults);
+        }
+    }
+    if(n_kwonlyargs > 0) {
+        _m_kwonlyargs.reserve(al, n_kwonlyargs);
+        defaults.reserve(al, n_kwonlyargs);
```

Comment:
You seem to be calling "reserve" couple times on `defaults`. That might be the issue.

Reply:
Replace this:
```

    Vec<arg_t> _m_args;
    _m_args.from_pointer_n(nullptr, 0);
```
with:
```
    Vec<arg_t> _m_args;
    _m_args.reserve(al, 4);
```
The same for all the other ones. Remove all calls to reserve beyond that. I don't know if this fixes the issue at the CI, but that's one issue I can see.

Reply:
Yup, I will do it.